src/cljs/athens/page.cljs

```diff
@@ -45,12 +45,12 @@
                 [render-blocks uid]])])))])))
 
 ; match [[title]] or #title or #[[title]]
+; provides groups useful for replacing
+; e.g.: $1$3$4new-string$2$5
 (defn linked-pattern [string]
-  (re-pattern (str "("
-                   "\\[{2}" string "\\]{2}"
-                   "|" "#" string
-                   "|" "#" "\\[{2}" string "\\[{2}"
-                   ")")))
+  (re-pattern (str "(\\[{2})" string "(\\]{2})"
+                   "|" "(#)" string
+                   "|" "(#\\[{2})" string "(\\]{2})")))
```

Comment:
This looks pretty complicated and asks for some examples via tests. Maybe you can extract this functionality into a cljc file and test this behaviour with `deftest` in `test/cljc/athens/page_test.cljc`?

Reply:
Could we use https://github.com/lambdaisland/regal ?

Reply:
@jeroenvandijk I've added some tests ann extracted the patterns to a new cljc file.

Reply:
> Could we use https://github.com/lambdaisland/regal ?

@tomisme, maybe we could. But it might be better to actually use instaparse and the existing grammar for parsing links for this. That way there is only one "pattern" that defines how a link should be detected in Athens.

@jeroenvandijk what do you think?

Reply:
Yeah regal could be nice at some point, especially when we want to do generative testing. But we don't need it yet, so I think the current changes are fine. Since we have many new clojure people here, it is also good to not make too many steps at once, people have an easier time to understand the steps. We can upgrade the regexp's to Regal later. 

Regarding instaparse, yes we need that for the markdown parsing at least. Instaparse supports regexp's as well. Maybe there is some overlap here. Let's wait and see.

(*) I personally also don't have experience with Regal yet, but I'm sure it's good.